iiod/responder.c

```diff
@@ -681,6 +697,24 @@ static void handle_create_block(struct parser_pdata *pdata,
 	entry->io = io;
 	entry->client_id = cmd->client_id;
 
+	if (WITH_IIOD_USB_DMABUF && pdata->is_usb) {
+		entry->dmabuf_fd = iio_block_get_dmabuf_fd(block);
+		if (entry->dmabuf_fd > 0) {
+			ep_fd = buf_entry->is_tx ? pdata->fd_in : pdata->fd_out;
+
+			ret = usb_attach_dmabuf(ep_fd, entry->dmabuf_fd);
+			if (!ret) {
+				/* We could attach to functionfs. Disable CPU
+				 * access to the block as we won't need it. */
+				iio_block_disable_cpu_access(block, true);
+			} else {
+				/* If we can't attach - no problem. The
+				 * data will be transferred the regular way. */
+				entry->dmabuf_fd = -EINVAL;
+			}
+		}
+	}
```

Comment:
nit: maybe we should already have a mechanism in here to properly handle this in the proper backend? For now, yes, only USB supports this but ideally we can have the same in net.

Maybe a simple helper function or a new callback? 

Not really a blocker from my point of but maybe something easy enough that could be done already instead of doing it when a new backend supports this mechanism

Reply:
USB does not support DMABUFs, the gadget/functionfs API does. So this is very much a IIOD-side things that shouldn't be in Libiio.

If we ever get DMABUF support to the network stack, the related code could be shared (as a static lib akin to iiod-responder) between the network backend and IIOD.

Reply:
> So this is very much a IIOD-side things that shouldn't be in Libiio.

But don't we have a similar situation if network is being used? IIOD still needs to "respond" the data right?

Reply:
Right. We could have a "frontend" system, but I'd prefer refactoring the code once we do have DMABUF support in the network stack.